@enkilee 

``` python
... print(type(img), img.size) # type: ignore
```

这里是不需要 ignore 的，而

``` python
... print(type(img), img.shape)
```

这里需要 ～

主要是因为 `Tensor, PILImage, ndarray` 都有 size 属性，但是 PILImage 没有 shape ，所以 mypy 检查的时候取类型中属性集合的交集就报错了～

mypy 会检查无用的 ignore，所以报错了 `<string>:42: error: Unused "type: ignore" comment  [unused-ignore]` ～

--- a/python/paddle/vision/datasets/folder.py
+++ b/python/paddle/vision/datasets/folder.py
@@ -14,7 +14,7 @@
 
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Any, Callable, Literal
+from typing import TYPE_CHECKING, Any, Callable, Literal, Sequence
 
 from typing_extensions import TypeAlias
 
@@ -47,9 +47,7 @@ from paddle.utils import try_import
 __all__ = []
 
 
-def has_valid_extension(
-    filename: str, extensions: list[str] | tuple[str, ...]
-) -> bool:
+def has_valid_extension(filename: str, extensions: Sequence[str]) -> bool:
     """Checks if a file is a valid extension.
 
     Args:
@@ -231,7 +229,7 @@ class DatasetFolder(Dataset):
     """
 
     loader: Callable[..., _ImageDataType] | None
-    extensions: _AllowedExtensions | None
+    extensions: Sequence[_AllowedExtensions] | None
     transform: _Transform[Any, Any] | None
     classes: list[str]
     class_to_idx: dict[str, int]
@@ -243,7 +241,7 @@ class DatasetFolder(Dataset):
         self,
         root: str,
         loader: Callable[..., _ImageDataType] | None = None,
-        extensions: _AllowedExtensions | None = None,
+        extensions: Sequence[_AllowedExtensions] | None = None,
         transform: _Transform[Any, Any] | None = None,
         is_valid_file: _ImageDataType | None = None,
     ) -> None:
@@ -467,7 +465,7 @@ class ImageFolder(Dataset):
     """
 
     loader: Callable[..., _ImageDataType] | None
-    extensions: _AllowedExtensions | None
+    extensions: Sequence[_AllowedExtensions] | None
     samples: list[str]
     transform: _Transform[Any, Any] | None
 
@@ -475,7 +473,7 @@ class ImageFolder(Dataset):
         self,
         root: str,
         loader: Callable[..., _ImageDataType] | None = None,
-        extensions: _AllowedExtensions | None = None,
+        extensions: Sequence[_AllowedExtensions] | None = None,
         transform: _Transform[Any, Any] | None = None,
         is_valid_file: _ImageDataType | None = None,
     ) -> None:
@@ -510,7 +508,7 @@ class ImageFolder(Dataset):
         self.samples = samples
         self.transform = transform
 
-    def __getitem__(self, index: int) -> tuple[_ImageDataType]:
+    def __getitem__(self, index: int) -> list[_ImageDataType]:
         """
         Args:
             index (int): Index

--- a/python/paddle/vision/datasets/folder.py
+++ b/python/paddle/vision/datasets/folder.py
@@ -209,7 +209,7 @@ class DatasetFolder(Dataset):
             >>> data_folder_2 = DatasetFolder(
             ...     fake_data_dir,
             ...     loader=lambda x: cv2.imread(x),  # load image with OpenCV
-            ...     extensions=(".jpg",),  # type: ignore # only load *.jpg files
+            ...     extensions=(".jpg",),  # only load *.jpg files
             ...     transform=transform,  # apply transform to every image
             ... )
 
@@ -222,7 +222,7 @@ class DatasetFolder(Dataset):
 
             >>> for img, label in iter(data_folder_2):
             ...     # do something with img and label
-            ...     print(type(img), img.shape, label) # type: ignore
+            ...     print(type(img), img.shape, label)  # type: ignore
             ...     # <class 'paddle.Tensor'> [3, 64, 64] 0
 
             >>> shutil.rmtree(fake_data_dir)
@@ -445,7 +445,7 @@ class ImageFolder(Dataset):
             >>> image_folder_2 = ImageFolder(
             ...     fake_data_dir,
             ...     loader=lambda x: cv2.imread(x),  # load image with OpenCV
-            ...     extensions=(".jpg",),   # type: ignore # only load *.jpg files
+            ...     extensions=(".jpg",),  # only load *.jpg files
             ...     transform=transform,  # apply transform to every image
             ... )
 
@@ -458,7 +458,7 @@ class ImageFolder(Dataset):
 
             >>> for (img,) in iter(image_folder_2):
             ...     # do something with img
-            ...     print(type(img), img.shape) # type: ignore
+            ...     print(type(img), img.shape)  # type: ignore
             ...     # <class 'paddle.Tensor'> [3, 64, 64]
 
             >>> shutil.rmtree(fake_data_dir)

--- a/python/paddle/vision/datasets/folder.py
+++ b/python/paddle/vision/datasets/folder.py
@@ -428,7 +428,7 @@ class ImageFolder(Dataset):
             >>> for i in range(len(image_folder_1)):
             ...     (img,) = image_folder_1[i]
             ...     # do something with img
-            ...     print(type(img), img.size) # type: ignore
+            ...     print(type(img), img.size)
             ...     # <class 'PIL.Image.Image'> (32, 32)
 
 
@@ -460,7 +460,7 @@ class ImageFolder(Dataset):
 
             >>> for (img,) in iter(image_folder_2):
             ...     # do something with img
-            ...     print(type(img), img.shape)
+            ...     print(type(img), img.shape) # type: ignore
             ...     # <class 'paddle.Tensor'> [3, 64, 64]
 
             >>> shutil.rmtree(fake_data_dir)

